AsyncDisplayKit/ASTableView.mm

```diff
@@ -903,6 +915,21 @@ - (void)rangeController:(ASRangeController *)rangeController didDeleteSectionsAt
     return; // if the asyncDataSource has become invalid while we are processing, ignore this request to avoid crashes
   }
 
+  if (_contentOffsetAdjustmentTopVisibleRow != nil) {
+    __block NSInteger adjustedSection = _contentOffsetAdjustmentTopVisibleRow.section;
+    __block NSInteger adjustedRow = _contentOffsetAdjustmentTopVisibleRow.row;
+    [indexSet enumerateIndexesUsingBlock:^(NSUInteger idx, BOOL *stop) {
+        if (adjustedSection > idx) {
+          adjustedSection--;
+        } else if (adjustedSection == idx) {
+          adjustedRow = 0;
```

Comment:
Things can get tricky here if there are sections with no rows. To do this really correctly we would need to detect zero-row sections and skip them (because in this case adjustedRow = 0 would be an invalid index path.) This could be done outside the enumerate block or perhaps in the else where we actually set adjustedRow to 0. I'm not sure offhand what the best approach is to get the valid (and safe) row count at this point in the code base. Unfortunately, zero length sections are a fairly common way to optionally hide content, so this could happen.